### PR TITLE
[Bug fixes] fix gpt tipic testing

### LINKED ISSUE #4819: [Bug]: TypeError: __init__() got an unexpected keyword argument 'vocab_size'
### 软件环境

```Markdown
- paddlepaddle:
- paddlepaddle-gpu: 2.4.1.post112
- paddlenlp: 2.5.0
```

### 重复问题

- [X] I have searched the existing issues

### 错误描述

```Markdown
But there is a bug:
/home/qyh/anaconda3/envs/py39/lib/python3.9/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
Traceback (most recent call last):
  File "/resources/qyh/big-model/PaddleNLP/model_zoo/gpt/run_eval.py", line 291, in <module>
    run()
  File "/resources/qyh/big-model/PaddleNLP/model_zoo/gpt/run_eval.py", line 287, in run
    do_eval(args)
  File "/resources/qyh/big-model/PaddleNLP/model_zoo/gpt/run_eval.py", line 231, in do_eval
    model = GPTForPretraining(GPTModel(**model_class.pretrained_init_configuration[args.model_name]))
  File "/home/qyh/.local/lib/python3.9/site-packages/paddlenlp/transformers/utils.py", line 170, in __impl__
    init_func(self, *args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'vocab_size'


How can I fix the issure?
```

### 稳定复现步骤 & 代码

I train with the following code:

CUDA_VISIBLE_DEVICES=7 python run_pretrain.py \
    --model_type gpt \
    --model_name_or_path gpt2-en \
    --input_dir "./data"\
    --output_dir "output"\
    --weight_decay 0.01\
    --grad_clip 1.0\
    --max_steps 500000\
    --save_steps 100000\
    --decay_steps 320000\
    --warmup_rate 0.01\
    --micro_batch_size 4\
    --device gpu

And eval with the following code:

CUDA_VISIBLE_DEVICES=6 python run_eval.py --model_name gpt2-en \
     --eval_path ./wikitext-103/wiki.valid.tokens \
     --overlapping_eval 32 \
     --init_checkpoint_path ./output/model_100000/model_state.pdparams \
     --batch_size 4 \
     --device gpu

### PR DESCRIPTION
**Pull request by wj-Mcat**
### PR types

Bug fixes

### PR changes

Testing

### Description

解决 gpt 在 tipc 训练中参数不匹配的问题

### CODE REVIEW

**Review by FrostML — DISMISSED** [review]

File: tests/test_tipc/configs/gpt2/train_infer_python.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@ null:null
 null:null
 ##
 trainer:norm_train
-norm_train:../examples/language_model/gpt/run_pretrain.py --model_type="gpt" --model_name_or_path="gpt2-en" --save_steps 100000 --decay_steps 320000 --weight_decay 0.01 --warmup_rate 0.01 --grad_clip 1.0 --logging_freq 1 --eval_freq 1000 --device "gpu" --min_lr 0.00001 --max_lr 0.00015 --max_seq_len 1024 --output_dir=./output/ --input_dir=./data/
+norm_train:../examples/language_model/gpt/run_pretrain.py --model_type="gpt" --model_name_or_path="gpt2-en"  --warmup_steps=320000 --warmup_ratio=0.01 --eval_steps=1000 --do_train=true --do_predict=true --device=gpu --max_seq_len=1024 --max_grad_norm=1.0 --output_dir=./output/ --input_dir=./data/ --save_steps=10000 --weight_decay=0.01 --logging_steps=1
```
Comment (ZHUI) [inline tests/test_tipc/configs/gpt2/train_infer_python.txt]: --do_predict=true 
```suggestion
norm_train:../examples/language_model/gpt/run_pretrain.py --model_type="gpt" --model_name_or_path="gpt2-en"  --warmup_steps=320000 --warmup_ratio=0.01 --eval_steps=1000 --do_train=true  --device=gpu --max_seq_len=1024 --max_grad_norm=1.0 --output_dir=./output/ --input_dir=./data/ --save_steps=100000 --weight_decay=0.01 --logging_steps=1
```

File: tests/test_tipc/benchmark_train.sh [hunk 2]
```diff
@@ -230,8 +230,7 @@ for batch_size in ${batch_size_list[*]}; do
                 num_gpu_devices=`get_world_size $device_num`
                 sed_norm_train=$norm_train
 
-                global_batch_size=$(($batch_size*$num_gpu_devices))
-                extra_params="--global_batch_size=$global_batch_size --dp_degree=$num_gpu_devices"
+                extra_params="--micro_batch_size=$batch_size --dp_degree=$num_gpu_devices"
```
Comment (ZHUI) [inline tests/test_tipc/benchmark_train.sh]:
```suggestion
                extra_params="--pre_device_device_train_batch_size =$batch_size --dp_degree=$num_gpu_devices"
```

File: model_zoo/gpt/run_pretrain.py [hunk 3]
```diff
@@ -53,6 +53,9 @@
 class TrainingArguments(TrainingArguments):
     min_lr: float = field(default=1e-5, metadata={"help": "The initial min learning rate for Adam."})
     micro_batch_size: int = field(default=4, metadata={"help": "the batch-size of trainin"})
```
Comment (ZHUI) [inline model_zoo/gpt/run_pretrain.py]: del

File: model_zoo/gpt/run_pretrain.py [hunk 4]
```diff
@@ -53,6 +53,9 @@
...
+    dp_degree: int = field(default=128, metadata={"help": "Data Parallelism degree."})
```
Comment (ZHUI) [inline model_zoo/gpt/run_pretrain.py]: None if paddle.disturibute.get_world_size

File: model_zoo/gpt/run_pretrain.py [hunk 5]
```diff
@@ -53,6 +53,9 @@
...
+    profiler_options: str = field(default="gpt", metadata={"help": "key1=value1;key2=value2;key3=value3"})
+    use_amp: bool = field(default=False, metadata={"help": "Enable mixed precision training."})
```
Comment (ZHUI) [inline model_zoo/gpt/run_pretrain.py]: fp16

File: model_zoo/gpt/README.md [hunk 6]
```diff
@@ -87,31 +87,35 @@ mv gpt_en_dataset_300m_idx.npz ./data
...
-  --eval_steps 100 \
-  --do_train true \
-  --do_predict true
+    --model_name_or_path gpt2-en \
+    --input_dir ./data \
+    --output_dir ./output_dir/pretrain \
+    --weight_decay 0.01 \
+    --max_steps 500000 \
+    --save_steps 100000 \
+    --warmup_steps 320000 \
```
Comment (ZHUI) [inline model_zoo/gpt/README.md]:
```suggestion
    --decay_steps 320000 \
```

这个参数是有用的
https://github.com/PaddlePaddle/PaddleNLP/blob/v2.4.9/model_zoo/gpt/args.py#L108-L113

File: model_zoo/gpt/README.md [hunk 7]
```diff
@@ -87,31 +87,35 @@ mv gpt_en_dataset_300m_idx.npz ./data
...
 - `input_dir` 指定输入文件，可以使用目录，指定目录时将包括目录中的所有文件。
 - `output_dir` 指定输出文件。
 - `weight_decay` 权重衰减参数。
-- `grad_clip` 梯度裁剪范围。
 - `max_steps` 最大训练步数
-- `save_steps` 保存模型间隔
-- `mirco_batch_size` 训练的batch大小
-- `device` 训练设备
+- `save_steps`  保存模型间隔
+- `warmup_steps` lr warmup 步数
```
Comment (ZHUI) [inline model_zoo/gpt/README.md]:
```suggestion

```

**Comment by wj-Mcat** [comment]
Fix https://github.com/PaddlePaddle/PaddleNLP/issues/4819

**Pull request closed without merging**